app/card-info/[id]/page.tsx

```diff
@@ -1,21 +1,56 @@
-// The detailed information of a card if just a whole seperate page with one card by itself
-// It is a solution made on the basis of non existent requirements for this page
-// So a developer went on the path of a least resistance
 "use client";
+
 import { useParams } from "next/navigation";
-import TodoCard from "@/app/components/TodoCard";
+import TodoDetails from "@/app/components/TodoDetails";
 import dataService from "@/services/dataService";
 import { TodoCardInfo } from "@/types";
+import { useEffect, useState } from "react";
 
 export default function CardDetailsPage() {
-  const params = useParams();
-  const { id } = params;
-  const oneCardData = dataService
-    .getData()
-    .filter((card: TodoCardInfo) => card.id === id)[0];
+  const { id } = useParams();
+  const idString: string = Array.isArray(id) ? id[0] : id;
```

Comment:
This approach indeed looks better. 
It is being checked as an array, because eslint says that in URL it is possible to have multiple values, so there is possibility that id could be an array, so I will give you an error that you can get rid of only by explicitly making sure that id is not an array

Reply:
Ah now I understand where you got that from. Multiple values is suppose to be ment like this: /endpoint?id=value1&id=value2 in the url but for this one standard routes like /endpoint/:id, id will always be a string.